### PR TITLE
Front end Role checking to limit Specializations

### PR DESCRIPTION
**Pull request by JustinMacaulay**
### Motivation and Context

Looks at the Microsoft Entra groups and limits the specializations to the ones associated to those groups.

### Description

Front end will now only show you specializations based on the groups you are associated with in entra.

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [Contribution Guidelines](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: webapi/Models/Storage/SpecializationSession.cs [hunk 1]
```diff
@@ -27,11 +29,17 @@ public class SpecializationSession
     /// </summary>
     public string ImageFilepath { get; set; }
 
-    public SpecializationSession(string Key, string Name, string Description, string ImageFilepath)
+    /// <summary>
+    /// Image URL for pictorial description of specialization or logo etc.
```
Comment (dhaselhan) [inline webapi/Models/Storage/SpecializationSession.cs:33]: nitpick: Text needs an update

**Review by dhaselhan — APPROVED** [review]
LGTM 👍

File: webapi/Models/Storage/SpecializationSession.cs (hunk 1 above)
Comment (JustinMacaulay) [reply webapi/Models/Storage/SpecializationSession.cs]: Fixed

**Review by dhaselhan — APPROVED** [review]
LGTM 👍

**Review by dhaselhan — APPROVED** [review]
LGTM 👍

**Pull request merged**